### DISCUSSION TITLE
Kata 7 duda

### DISCUSSION THREAD

**Discussion by JhosianGtz** [LaunchX-InnovaccionVirtual/CursoIntroPython#219] [Q&A] [answered]
### porque ejecute el programa  sin el y funcionó igual.

![image](https://user-images.githubusercontent.com/40299796/153768594-9f2d976f-36e1-4f87-8a02-7666358726f8.png)

**Comment by jstoledano** [comment]
Para que si `newPlanet `es diferente de `done`, lo agregue a la lista `planets`. Si quitas solo la línea del `if`, seguramente va a seguir funcionando, porque cualquier cosa que pongas se va a agregar a lista.

Pero la intención de este patrón es agregar validaciones un poco más complicadas, como por ejemplo, asegurar que el nombre del planeta tenga menos de tres caracteres o que no exista previamente en la lista.

**Reply by JhosianGtz** [reply]
### Pero ese if en especifico ¿Que valida?

**Reply by JhosianGtz** [reply]
### Porque el while valida que newPlanet sea diferente de done, luego pasa al if ¿pero que hace ahi?

**Reply by ohmesz** [reply]
valida que la cadena de `newPlanet` contenga algun valor

**Reply by JhosianGtz** [reply]
### Pero lo valida sin ningun operador?

**Reply by JhosianGtz** [reply]
### No debería ir así?

![image](https://user-images.githubusercontent.com/40299796/153769615-78c07642-3faf-4fed-967f-609bf3b20376.png)

**Comment by JhosianGtz** [comment, marked as answer]
### Ya entendí por qué no lleva un operador, gracias por las repuestas.

### Acá hay un ejemplo de validación:
https://parzibyte.me/blog/2019/01/18/comprobar-cadena-vacia-python/
